automation.md

```diff
@@ -0,0 +1,75 @@
+
+## Setup
+
+If you chose *Service Principal* authentication, you need to create an Active Directory application with the rights to create and remove resources in the desired Azure subscription. To achieve this, create a new application in Azure Active Directory and then navigate to "Subscriptions &rarr; [Subscription Id] &rarr; Access Control (IAM) &rarr; Role Assignments" and then add the newly created AAD Application with "Contributor" role.
+
+If you chose *Managed Identity* authentication, the application will run with the permissions of the currently logged in user or the Azure DevOps pipeline's permissions. Make sure that you are logged in to the proper Azure account using the [AZ command line interface tool](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest).
+
+## Configuration
+
+The configuration for the automation component is kept in [appsettings.json](/src/IotTelemetrySimulator.Automation/appsettings.json) or for local development purposes in **appsettings.Development.json**. It contains the following fields
+
+|Parameter|Description|Remarks
+|-|-|-|
+|AuthenticationMethod|*Service Principal* or *Managed Identity*|
+|AadClientId|Azure Active Directory app registration Application (Client) Id|Only used with Service Principal authentication method|
+|AadClientSecret|Azure Active Directory app registration Client Secret|Only used with Service Principal authentication method|
+|AzureSubscriptionName|Azure Subscription Name|Only used with Service Principal authentication method|
+|AadTenantId|Azure Active Directory Tenant Id (Get by navigating to "Azure &rarr; Active Directory &rarr; Manage &rarr; Properties &rarr; Directory ID")|Used with Service Principal and Managed Identity authentication methods|
+|AzureSubscriptionId|Azure Subscription Id|Only used with Managed Identity authentication method|
+|AzureRegion|Azure Region as formatted in the [Microsoft.Azure.Management.ResourceManager.Fluent.Core.Region Class' Fields](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.management.resourcemanager.fluent.core.region?view=azure-dotnet)|All resources will be created in this region.|
+|ResourceGroupName|The name of the Azure Resource Group created by to host the Container Groups|
+|ContainerGroupName|The name prefix under which the Azure Container Groups will be created|
+|ContainerImage|The full path to the docker image to be pulled by the Azure Container Instances. For example: `acrname.azurecr.io/azureiot-telemetrysimulator:latest`|The automation component has been configured to pull the image from an Azure Container Registry.
+|AcrServer|The name of the Azure Container Registry. I.e. `acrname.azurecr.io`|
+|AcrUsername|The name of user with which to access the Azure Container Registry|
+|AcrPassword|The above user's Azure Container Registry password|
```

Comment:
Can I still use the image from docker hub (fbeltrao/xxx)?
Having to create an own image on ACR might be useful in some projects, but it is an additional step that might not be required.

Reply:
I'll add the feature. It's a bit of work.